generators/base/generator-base-private.mjs

```diff
@@ -465,40 +220,15 @@ export default class PrivateBase extends Generator {
    * @param {string[]} args - arguments to print
    */
   debug(msg, ...args) {
-    const formattedMsg = `${chalk.yellow.bold('DEBUG!')} ${msg}`;
-    if ((this.configOptions && this.configOptions.isDebugEnabled) || (this.options && this.options.debug)) {
-      this.log(formattedMsg);
-      args.forEach(arg => this.log(arg));
-    }
-    if (this._debug && this._debug.enabled) {
-      this._debug(formattedMsg);
-      args.forEach(arg => this._debug(arg));
-    }
+    logDebug(this, msg, ...args);
   }
 
   /**
    * @private
    * Check if Java is installed
    */
   checkJava() {
```

Comment:
We don't need to have checkJava at base generator.

Reply:
I'll move it later: for now it's used by two generators so it's not that simple to extract.

Reply:
removed in favour of your implementation